pandas/core/computation/ops.py

```diff
@@ -197,7 +197,9 @@ class Op:
     Hold an operator of arbitrary arity.
     """
 
-    def __init__(self, op, operands, *args, **kwargs):
+    op: str
```

Comment:
is this a typo?

Reply:
no, there is a `op` attribute that is a str

Reply:
is it necessary?

```
reveal_type(_bool_op_map)
reveal_type(_bool_op_map.get("not", "not"))
```
```
pandas\core\computation\ops.py:193: note: Revealed type is 'builtins.dict[builtins.str*, builtins.str*]'
pandas\core\computation\ops.py:194: note: Revealed type is 'builtins.str'
```

Reply:
> is it necessary?

i guess im not clear on the policy.  i thought it was a "more is better" kind of thing

Reply:
I'm just suspicious when I see unnecessary variable annotations since they could be casts in disguise. 

In this case, it _could_ have been (but wasn't) hiding that the return type of _bool_op_map.get(op,op) might be Any. 

If the return type of some_function is Any and we do `some variable: some_type  = some_function()` it is effectively a cast.

```python
def some_function():
    return "foo"

x: int = some_function()
reveal_type(x)
```
```
test.py:5: note: Revealed type is 'builtins.int'
```
I think we should just allow mypy to use the return type of the initialization unless needed.